setup.py

```diff
@@ -262,18 +261,20 @@ def run(self):
     'pytest-rerunfailures',
     'mock',
     'cython',
-    # temporarily remove pyemd to work around appveyor issues
-    # 'pyemd',
+    'nmslib',
```

Comment:
in case if you change dependencies here, you should "mirror" this changes in `gensim-wheels`

Reply:
Why? How is this related to wheels?

Reply:
though testing of course, if you change test deps here, you should use "almost same" in wheel testing too

Reply:
I don't know how that works – can I leave `gensim-wheels` with you?

Reply:
LOL I guess. Is the process documented somewhere?

And most importantly: is it worth it? How many actual bugs did that catch, versus how much upkeep & maintenance effort.

Reply:
Created #2816 to deal with the issue outside of this PR.

Reply:
>  Is the process documented somewhere? 

https://github.com/matthew-brett/multibuild

> is it worth it? 

Generally yes, because of this catch compilation issues / wired behavior on all versions of all OS (what we don't do in current repo)

> How many actual bugs did that catch over the years, versus how much upkeep & maintenance effort.

Depends on "actual bug" definition, maintenance effort correlated with `gensim` changes (like new test dependencies, new cython/c++ deps, new python version, etc). It burned a lot of time, that's true. I want to believe it's not in vain.

Reply:
What's an example of a bug this has caught? First / most important that comes to mind.

Reply:
Compilation issues on OSX & Windows (esp. windows + py27). This appears multiple times in cases when we add new python version support or new `pyx`/`c`/`cpp`

Reply:
I'd expect compilation issues to be caught by CI.

Missing files in wheels could be a problem.

@mpenkov I'll look into that `multibuild` thing (or maybe the simpler `cibuildwheel` linked above) eventually, but not blocking for us now.

I'll aim to drop and simplify as much as possible. If the complexity – combined with the severe lack of documentation – is such that it leads to contributor frustration ("constantly repairing your wheels, to be honest, I'm sick of it") then that's not sustainable for a volunteer open source project.

Reply:
> I'd expect compilation issues to be caught by CI.

Not really, because
- fully missing OSX
- missing some of win+py combinations

In wheel building, we checked almost all stuff

Reply:
btw still waiting https://github.com/RaRe-Technologies/gensim/pull/2814#issuecomment-621950805